app/components/submission-status-display/component.js

```diff
@@ -0,0 +1,12 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+  submissionHandler: service('submission-handler'),
+
+  status: computed('record', 'submission', function () {
+    const submission = this.get('submission') || this.get('record');
```

Comment:
use the imported get where possible, if you need a get. Does `this.submission` return a promise-y thing?

Reply:
Nope, this was likely just old habits showing up